### PR TITLE
Routing for deployment modal

### PR DESCRIPTION
**Pull request by halkyon**

### CODE REVIEW

File: js/containers/DeployModal.jsx [hunk 1]
```diff
@@ -112,6 +111,16 @@ const mapStateToProps = function(state) {
...
 	}
 
+	let active_step = 0;
+	if (window.location.hash) {
+		active_step = parseInt(window.location.hash.substring(1), 10);
+	}
+
+	let is_open = false;
+	if (ownProps.params) {
+		is_open = ownProps.params.id !== null || typeof (ownProps.params.id) !== 'undefined';
```
Comment (stojg) [inline js/containers/DeployModal.jsx]: you are doing a three combination comparison if here, just hidden as one `if` and a double `true comparison` assignment. Change to either use only an if or the comparison to make it more readable.
```
if (typeof (ownProps.params.id) !== 'undefined' && ownProps.params.id !== null) {
   is_open = true;
}
```
or 

```
is_open = typeof (ownProps.params.id) !== 'undefined' && ownProps.params.id !== null;
```

File: js/_actions.js [hunk 2]
```diff
@@ -6,6 +6,8 @@ const gitAPI = api.create('git');
 const planAPI = api.create('plan');
 const deployAPI = api.create('deploys');
 
+export const history = null;
```
Comment (halkyon) [inline js/_actions.js]: yeah, it's an exported variable that is set by EnvironmentOverview.jsx. I'll add some comments

File: js/EnvironmentOverview.jsx [hunk 3]
```diff
@@ -1,33 +1,47 @@
...
+const store = Redux.createStore(
+	reducers,
 	Redux.applyMiddleware(
 		thunkMiddleware,
 		createLogger()
 	)
 );
 
-var Plan = function(props) {
+export function getDeploymentHook(store) {
```
Comment (halkyon) [inline js/EnvironmentOverview.jsx]: yeah, that's right. Any ideas on the name? It's basically taking the URL param and getting the deployment (or setting a new deployment).

File: code/control/EnvironmentOverview.php [hunk 4]
```diff
@@ -16,6 +16,10 @@ class EnvironmentOverview extends Dispatcher {
 		self::ACTION_OVERVIEW
 	];
 
+	private static $allowed_actions = [
```
Comment (halkyon) [inline code/control/EnvironmentOverview.php]: not wrong, I'll document

File: js/EnvironmentOverview.jsx (hunk 3 above)
Comment (halkyon) [inline js/EnvironmentOverview.jsx]: `loadDeployment`?

**Pull request merged**